app/models/katello/content_view.rb

```diff
@@ -579,14 +579,16 @@ def check_docker_repository_names!(environments)
 
     def check_composite_action_allowed!(env)
       if composite? && Setting['restrict_composite_view']
-        # verify that the composite's component view versions exist in the target environment.
+        env_ids = env.try(:pluck, 'id') || []
+        env_ids << env.id unless env_ids.size > 0
         components.each do |component|
-          unless component.environments.include?(env)
+          component_environment_ids = component.environments.try(:pluck, 'id')
+          unless (env_ids - component_environment_ids).empty?
             fail _("The action requested on this composite view cannot be performed until all of the "\
                    "component content view versions have been promoted to the target environment: %{env}.  "\
                    "This restriction is optional and can be modified in the Administrator -> Settings "\
                    "page using the restrict_composite_view flag.") %
-                   { :env => env.name }
+                   { :env => env.try(:pluck, 'name') || env.name }
```

Comment:
why not `env.pluck(:name).join(', ')`? (for the same reasons as above)

Reply:
Same reason as above. Let me know if it sounds correct to you..